optuna/integration/sklearn.py

```diff
@@ -237,10 +300,18 @@ def __call__(self, trial: Trial) -> float:
                 return_train_score=self.return_train_score,
                 scoring=self.scoring,
             )
+            self.multimetric_ = "test_score" not in scores
+
+            # check refit_metric now for a callabe scorer that is multimetric
+            if callable(self.scoring) and self.multimetric_:
+                _check_refit_for_multimetric(scores, self.refit)
 
         self._store_scores(trial, scores)
 
-        return trial.user_attrs["mean_test_score"]
+        if self.multimetric_:
+            return trial.system_attrs[f"mean_test_{self.refit}"]
```

Comment:
I may not be familiar with optuna codebase, but is the key "mean_test_score" reserved?
Otherwise, for sake of users familiar with sklearn, it might be a better option to keep the key as same as sklearn.

Reply:
Thank you for asking the question. Users might use `system_attrs` or `user_attrs` to store their own metrics during optimization. For me, `mean_test_score` sounds like a generic name, so users use the same metric name to store for their own metric. In addition, the current implementation may not expect that the attributes are accessed by users because there is no information in Optuna docs. So I came up with the following two ways:

1. Adding a prefix as I said before.
2. Not adding a prefix, but adding a note section to explain the attributes.